src/Products/PluginIndexes/common/UnIndex.py

```diff
@@ -304,6 +346,69 @@ def _apply_not(self, not_parm, resultset=None):
     def _convert(self, value, default=None):
         return value
 
+    def _catalog_cache_key(self, catalog):
+        cid = catalog.getId()
+        counter = getattr(aq_base(catalog), 'getCounter', None)
```

Comment:
getCounter is only implemented in Plone's CatalogTool subclass at https://github.com/plone/Products.CMFPlone/blob/master/Products/CMFPlone/CatalogTool.py#L367

It gets incremented whenever either catalog_object or uncatalog_object are called on the tool.

That logic should be moved to ZCatalog before it gets used here.